commons-db/src/main/java/com/palantir/nexus/db/sql/SqlConnectionHelper.java

```diff
@@ -0,0 +1,622 @@
+        updateMany(c, sql.getKey(), list);
+    }
+
+    boolean insertOneUnregisteredQuery(Connection c, String sql, Object... vs)
+            throws PalantirSqlException {
+        final int updated = insertOneCountRowsUnregisteredQuery(c, sql, vs);
+        assert updated == 1 : "expected 1 update, got : " + updated; //$NON-NLS-1$
+        return true;
+    }
+
+    boolean insertOne(Connection c, String key, Object... vs) throws PalantirSqlException {
+        final int updated = insertOneCountRows(c, key, vs);
+        assert updated == 1 : "expected 1 update, got : " + updated; //$NON-NLS-1$
+        return true;
+    }
+
+    boolean insertOne(Connection c, RegisteredSQLString sql, Object... vs)
+            throws PalantirSqlException {
+        return insertOne(c, sql.getKey(), vs);
+    }
+
+    int insertOneCountRowsUnregisteredQuery(Connection c, String sql, Object... vs)
+            throws PalantirSqlException {
+        return basicSql.insertOneCountRowsInternal(c, SQLString.getUnregisteredQuery(sql), vs);
+    }
+
+    int insertOneCountRows(Connection c, String key, Object... vs) throws PalantirSqlException {
+        return basicSql.insertOneCountRowsInternal(c, SQLString.getByKey(key, c), vs);
+    }
+
+    int insertOneCountRows(Connection c, RegisteredSQLString sql, Object... vs)
+            throws PalantirSqlException {
+        return insertOneCountRows(c, sql.getKey(), vs);
+    }
+
+    boolean insertManyUnregisteredQuery(Connection c, String sql, Iterable<Object[]> list)
+            throws PalantirSqlException {
+        return basicSql.insertMany(
+                c,
+                SQLString.getUnregisteredQuery(sql),
+                Iterables.toArray(list, Object[].class));
+    }
+
+    boolean insertMany(Connection c, String key, Iterable<Object[]> list)
+            throws PalantirSqlException {
+        return basicSql.insertMany(
+                c,
+                SQLString.getByKey(key, c),
+                Iterables.toArray(list, Object[].class));
+    }
+
+    boolean insertMany(Connection c, RegisteredSQLString sql, Iterable<Object[]> list)
+            throws PalantirSqlException {
+        return insertMany(c, sql.getKey(), list);
+    }
+
+    private final String SQL_TRUNCATE_PT_TEMP_IDS = " TRUNCATE TABLE /* SQL_TRUNCATE_PT_TEMP_IDS */ {0} "; //$NON-NLS-1$
+
+    private final String SQL_DELETE_PT_TEMP_IDS = " DELETE /* SQL_DELETE_PT_TEMP_IDS */ FROM {0} "; //$NON-NLS-1$
+
+    private final String SQL_DELETE_IDS_FROM_PT_TEMPS_IDS = " DELETE /* SQL_DELETE_IDS_FROM_PT_TEMPS_IDS */ FROM {0} WHERE id = ? "; //$NON-NLS-1$
+
+    // for large tables this may be faster, and if we know that pt_temp_ids is large, then we
+    // may want to use it. However, we've found that DELETE is faster on Oracle.
+    // NOTE: this query cannot be used with HSQLDB
+    // private final String SQL_TRUNCATE_PT_TEMP_IDS =
+    // " TRUNCATE /* SQL_TRUNCATE_PT_TEMP_IDS */ TABLE {0} ";
```

Comment:
stipp did performance testing on what the cutover point should be between these behaviors, if someone cares I can dig it up and we can put the logic in that is best